android/src/main/java/io/sentry/react/MapConverter.java

```diff
@@ -0,0 +1,95 @@
+            WritableArray writable = Arguments.createArray();
+            for (Object item : (ArrayList<?>) serialized) {
+                addValueToWritableArray(writable, convertToWritable(item));
+            }
+            return writable;
+        } else if (serialized instanceof Map) {
+            WritableMap writable = Arguments.createMap();
+            for (Map.Entry<?, ?> entry : ((Map<?, ?>) serialized).entrySet()) {
+                Object key = entry.getKey();
+                Object value = entry.getValue();
+
+                if (key instanceof String) {
+                    addValueToWritableMap(writable, (String) key, convertToWritable(value));
+                } else {
+                    logger.log(SentryLevel.ERROR, "Only String keys are supported in Map.", key);
+                }
+            }
+            return writable;
+        } else if (serialized instanceof Number
+            || serialized instanceof Boolean
+            || serialized == null
+            || serialized instanceof String) {
+            return serialized;
+        } else {
+            logger.log(SentryLevel.ERROR, "Supplied serialized value could not be converted.", serialized);
+            return null;
+        }
+    }
+
+    private static void addValueToWritableArray(WritableArray writableArray, Object value) {
+        if (value == null) {
+            writableArray.pushNull();
+        } else if (value instanceof Boolean) {
+            writableArray.pushBoolean((Boolean) value);
+        } else if (value instanceof Double) {
+            writableArray.pushDouble((Double) value);
+        } else if (value instanceof Integer) {
+            writableArray.pushInt((Integer) value);
+        } else if (value instanceof String) {
+            writableArray.pushString((String) value);
+        } else if (value instanceof ReadableMap) {
+            writableArray.pushMap((ReadableMap) value);
+        } else if (value instanceof ReadableArray) {
+            writableArray.pushArray((ReadableArray) value);
+        } else {
+            logger.log(SentryLevel.ERROR,
+                    "Could not convert object: " + value);
+        }
+    }
+
+    private static void addValueToWritableMap(WritableMap writableMap, String key, Object value) {
+        if (value == null) {
+            writableMap.putNull(key);
+        } else if (value instanceof Boolean) {
+            writableMap.putBoolean(key, (Boolean) value);
+        } else if (value instanceof Double) {
+            writableMap.putDouble(key, (Double) value);
+        } else if (value instanceof Integer) {
+            writableMap.putInt(key, (Integer) value);
+        } else if (value instanceof String) {
+            writableMap.putString(key, (String) value);
+        } else if (value instanceof ReadableArray) {
+            writableMap.putArray(key, (ReadableArray) value);
+        } else if (value instanceof ReadableMap) {
+            writableMap.putMap(key, (ReadableMap) value);
```

Comment:
@markushi Is there any other value that could come in the serialized Scope?

Reply:
Looks good! There's also `Float` and `Number` in Java world. More complex parts are all translated to primitives or strings, see the following test: https://github.com/getsentry/sentry-java/blob/feat/hybrid-scope-sync/sentry/src/test/java/io/sentry/util/MapObjectWriterTest.kt#L45-L110